src/drivers/roboclaw/roboclaw_params.c

```diff
@@ -0,0 +1,134 @@
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file roboclaw_params.c
+ *
+ * Parameters defined by the Roboclaw driver.
+ *
+ * The Roboclaw will need to be configured to match these parameters. For information about configuring the
+ * Roboclaw, see http://downloads.ionmc.com/docs/roboclaw_user_manual.pdf
+ *
+ * @author Timothy Scott <timothy@auterion.com>
+ */
+
+
+/**
+ * Uart write period
+ *
+ * How long to wait, in Milliseconds, between writing actuator controls over Uart to the Roboclaw
+ * @unit ms
+ * @min 1
+ * @max 1000
+ * @group Roboclaw driver
+ */
+PARAM_DEFINE_INT32(RBCLW_WRITE_PER, 10);
+
+/**
+ * Encoder read period
+ *
+ * How long to wait, in Milliseconds, between reading wheel encoder values over Uart from the Roboclaw
+ * @unit ms
+ * @min 1
+ * @max 1000
+ * @group Roboclaw driver
+ */
+PARAM_DEFINE_INT32(RBCLW_READ_PER, 10);
+
+/**
+ * Encoder counts per revolution
+ *
+ * Number of encoder counts for one revolution. The roboclaw treats analog encoders (potentiometers) as having 2047
+ * counts per rev. The default value of 1200 corresponds to the default configuration of the Aion R1 rover.
+ * @min 1
+ * @group Roboclaw driver
+ */
+PARAM_DEFINE_INT32(RBCLW_COUNTS_REV, 1200);
+
+/**
+ * Roboclaw serial baud rate
+ *
+ * Baud rate of the serial communication with the Roboclaw. The Roboclaw must be configured to match this rate.
+ * @min 1
+ * @max 8
+ * @value 1 2400 baud
+ * @value 2 9600 baud
+ * @value 3 19200 baud
+ * @value 4 38400 baud
+ * @value 5 57600 baud
+ * @value 6 115200 baud
+ * @value 7 230400 baud
+ * @value 8 460800 baud
+ * @group Roboclaw driver
+ */
+PARAM_DEFINE_INT32(RBCLW_BAUD, 8);
```

Comment:
Does the default here match the roboclaw default? And do we ever need to use another baudrate? We're usually trying not to have too many parameters until we really need them.

Reply:
It does not match the default, so I can change that. However, I think it is important to keep this as a parameter, because depending on the microcontroller that PX4 is running on, some baud rates might not be available, or higher baud rates might take too much CPU time (if no hardware serial is available).

Reply:
Also, it is important to increase the baud rate when possible, because at the default 2400 baud, the total of 42 bytes for each communication cycle can only happen about 7 times per second.